contentcuration/contentcuration/view/node_views.py

```diff
@@ -130,6 +130,32 @@ def get_nodes_by_ids(request):
                            .defer('node_id', 'original_source_node_id', 'source_node_id', 'content_id', 'original_channel_id', 'source_channel_id', 'source_id', 'source_domain', 'created', 'modified')
         return HttpResponse(JSONRenderer().render(ContentNodeSerializer(nodes, many=True).data))
 
+def get_node_path(request):
+    if request.method == 'POST':
+        data = json.loads(request.body)
+
+        try:
+            topic = ContentNode.objects.prefetch_related('children').get(node_id=data['topic_id'], tree_id=data['tree_id'])
+
+            if topic.kind_id != content_kinds.TOPIC:
+                node =  ContentNode.objects.prefetch_related('files')\
+                                            .prefetch_related('assessment_items')\
+                                            .prefetch_related('tags').get(node_id=data['topic_id'], tree_id=data['tree_id'])
+                nodes = node.get_ancestors(ascending=True)
+            else:
+                node =  data['node_id'] and ContentNode.objects.prefetch_related('files')\
+                                            .prefetch_related('assessment_items')\
+                                            .prefetch_related('tags').get(node_id=data['node_id'], tree_id=data['tree_id'])
+                nodes = topic.get_ancestors(include_self=True, ascending=True)
```

Comment:
My only suggestion for optimization would be to load this lazily after the fact, as this call can be slow. However, that would require a larger refactor than desirable at this point!